install.php

```diff
@@ -207,6 +247,185 @@ public function installOrUpdate($parent)
 	}
 
 	/**
+	 * Method to process SQL updates previous to the install process
+	 *
+	 * @param   object  $parent  Class calling this method
+	 *
+	 * @return  boolean          True on success
+	 */
+	public function preprocessUpdates($parent)
+	{
+		$manifest  = $parent->get('manifest');
+
+		if (isset($manifest->update))
+		{
+			if (isset($manifest->update->attributes()->folder))
+			{
+				$path = $manifest->update->attributes()->folder;
+
+				if (isset($manifest->update->pre) && isset($manifest->update->pre->schemas))
+				{
+					$schemapaths = $manifest->update->pre->schemas->children();
+
+					if (count($schemapaths))
+					{
+						$sourcePath = $parent->getParent()->getPath('source');
+
+						// If it just upgraded redCORE to a newer version using RFactory for database, it forces using the redCORE database drivers
+						if (substr(get_class(JFactory::$database), 0, 1) == 'J' && $this->extensionElement != 'com_redcore')
+						{
+							RFactory::$database = null;
+							JFactory::$database = null;
+							JFactory::$database = RFactory::getDbo();
```

Comment:
RFactory::getDbo() does it
https://github.com/redCOMPONENT-COM/redCORE/blob/develop/libraries/redcore/factory/factory.php#L58